File: flight_cheapest_date_search/v1/get/Node SDK/flight_cheapest_date_search.js
const Amadeus = require("amadeus");

const amadeus = new Amadeus({
  clientId: "YOUR_API_KEY",
  clientSecret: "YOUR_API_SECRET",
});

async function main() {
  try {
    // Find cheapest dates from Madrid to Munich
    const response = await amadeus.shopping.flightDates.get({
      origin: "MAD",
      destination: "MUC",
    });

    console.log(response);
  } catch (error) {
    console.error(error);
  }
}

main();

File: flight_cheapest_date_search/v1/get/Python SDK/flight_cheapest_date_search.py
# Install the Python library from https://pypi.org/project/amadeus
from amadeus import Client, ResponseError

amadeus = Client(
    client_id='YOUR_AMADEUS_API_KEY',
    client_secret='YOUR_AMADEUS_API_SECRET'
)

try:
    '''
    Find cheapest dates from Madrid to Munich
    '''
    response = amadeus.shopping.flight_dates.get(origin='MAD', destination='MUC')
    print(response.data)
except ResponseError as error:
    raise error
